Update docs to cover uv managed installations (#2072)
plone/documentation: docs/conceptual-guides/package-management.md

```

## Manage backend Python packages

### pip

> By convention in the Python community, {term}`pip` is commonly used to install Python packages. It is one supported way to install the Plone backend.

Reviewer: ```suggestion
By convention in the Python community, {term}`pip` is commonly used to install Python packages.
It is one supported way to install the Plone backend.
```

Each Plone version requires specific versions of many different packages.
So, pip should be used with constraints to make sure the correct versions are installed.
For example:

```shell
bin/pip install -c https://dist.plone.org/release/6.1-latest/constraints.txt Plone
```

In the Plone community, constraints are sometimes called "version pins."

> As a best practice, pip should always be used inside a specific {term}`virtualenv` to keep the packages separate from other applications.

Reviewer: ```suggestion
As a best practice, pip should always be used inside a specific Python {term}`virtual environment` to keep the packages separate from other applications.
```

(manage-packages-mxdev-label)=

### mxdev

During development, it is sometimes necessary to override the Plone version constraints.
This makes it possible to:
- install a newer version of a core Plone package that was released with a bugfix
- install an unreleased core Plone package from a source control system

> Unfortunately `pip` does not allow overriding constraints this way.

Reviewer: ```suggestion
Unfortunately pip does not allow overriding constraints this way.
```

{doc}`/admin-guide/add-ons`
```

### uv

More recently, {term}`uv` has become popular as a way to install Python packages.
This package manager is popular for its speed, its ability to manage the installation of Python itself, and its ability to consistently reproduce installed packages using a {file}`uv.lock` file.

> When a project is fully managed using `uv`, it is configured in `pyproject.toml` and the packages are installed using `uv sync`.

Reviewer: ```suggestion
When a project is fully managed using uv, it is configured in `pyproject.toml` and the packages are installed using `uv sync`.
```

> `uv` also has a backwards-compatible mode which works more like `pip`, and installs packages into a virtualenv using `uv pip install`.

Reviewer: ```suggestion
uv also has a backwards-compatible mode which works more like pip, and installs packages into a virtual environment via the command `uv pip install`.
```

> If you create a Plone project using Cookieplone, it creates a backend managed by `uv` (starting in March 2026).

Reviewer: Also align with Cookieplone version per above comment and `versionchanged` admonition.

```suggestion
If you create a Plone project using Cookieplone, it creates a backend managed by uv.
```

Author: You don't think the clarification about when that became effective is helpful?

Reviewer: I do, but see https://github.com/plone/documentation/pull/2072#issuecomment-4132204551.

### buildout

{term}`Buildout` is a tool for installing Python packages that has been used in the Plone community since about 2007, and is still preferred by some members of the community.

It not only installs packages, but can set up other things using an extensible system of "recipes."

> Buildout does not install Python packages into a virtualenv.

Reviewer: ```suggestion
Buildout does not install Python packages into a virtual environment.
```

Instead, it creates scripts that add the necessary packages to `sys.path` before running the script target.

## Manage frontend Node.js packages

### pnpm

Plone uses {term}`pnpm` to install Node.js packages.

> Compared to the standard {term}`NPM`, it has features that help with developing multiple Node.js packages in the same workspace.

Reviewer: Brand name is lowercase.

```suggestion
Compared to the standard {term}`npm`, it has features that help with developing multiple Node.js packages in the same workspace.
```

Author: I was following the capitalization I saw in the glossary.

Author: I fixed it in the glossary in https://github.com/plone/documentation/pull/2072/commits/0f5e395a5ec5252743d1ff4dd19e43974f0e14b6
